paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -42,18 +46,33 @@ import java.util.Locale
 class PaparazziPlugin : Plugin<Project> {
   @OptIn(ExperimentalStdlibApi::class)
   override fun apply(project: Project) {
-    require(project.plugins.hasPlugin("com.android.library")) {
-      "The Android Gradle library plugin must be applied before the Paparazzi plugin."
+    project.plugins.withId("com.android.library") {
+      setup(project, project.extensions.getByType(LibraryExtension::class.java).libraryVariants)
     }
+    project.plugins.withId("com.android.application") {
+      error(
+        "The Android Gradle Plugin must be for a library module. " +
+                "See https://github.com/cashapp/paparazzi/issues/107."
+      )
+    }
```

Comment:
can we move this to a separate PR, since this is about fixing the ordering problem?